components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/CacheBackedApplicationDAO.java

```diff
@@ -456,6 +485,12 @@ public void updateApplicationByResourceId(String resourceId,
         ServiceProvider storedApp = getApplicationByResourceId(resourceId, tenantDomain);
         clearAllAppCache(storedApp, tenantDomain);
 
+        // Clear the trusted app cache only if the trusted app metadata is changed.
+        if ((storedApp.getTrustedAppMetadata() != null &&
+                !storedApp.getTrustedAppMetadata().equals(updatedApp.getTrustedAppMetadata())) ||
+                (storedApp.getTrustedAppMetadata() == null && updatedApp.getTrustedAppMetadata() != null)) {
+            clearTrustedAppCache();
```

Comment:
Wondering whether we should go down another level and optimise the cache clearing. i.e. we essentially have two caches, iOS and Android. One way to optimise this is check if is fido trusted has changed then clear both caches otherwith check the relevant data change for iOS and android and do the cache clearing based on that.